src/Containers/App/App.js

```diff
@@ -13,35 +13,58 @@ class App extends Component {
   } 
 
   componentDidMount() {
-    this.generateColors()
+    this.checkLockedColors()
   }
 
-  generateColors = () => {
+  checkLockedColors = () => {
+    let currentPalette = []
+    if (this.props.currentPalette.length) {
+      this.props.currentPalette.map(colorObject => {
+        if (colorObject.locked) {
+          currentPalette.push(colorObject)
+        } else {
+          colorObject.hex =  '#' + Math.random().toString(16).slice(2, 8).toUpperCase()
+          colorObject = {hex: colorObject.hex, locked: colorObject.locked}
+          currentPalette.push(colorObject)
+        }
+        this.props.addPalette(currentPalette)
+      })
+    } else {
+      this.generateNewPalette()
+    }
+  }
+
+  generateNewPalette = async () => {
     let currentPalette = [];
     let currentColorObject = {
       locked: false
     }
-    for (var i = 0; i < 5; i++) {
+    for(let i = 0; i < 5; i++) {
       let hex = '#' + Math.random().toString(16).slice(2, 8).toUpperCase()
       currentPalette.push({hex, ...currentColorObject})
     }
-    this.props.addPalette(currentPalette)
+    await this.setState({ currentPalette })
```

Comment:
Yes.  I like how we are using both local state and the store.  Seems like a good way to organize things.